mirrord/intproxy/src/proxies/outgoing.rs

```diff
@@ -233,12 +335,59 @@ impl OutgoingProxy {
         session_id: LayerId,
         request: OutgoingConnectRequest,
         message_bus: &mut MessageBus<Self>,
-    ) {
-        self.queue(request.protocol)
-            .push_back(message_id, session_id);
+    ) -> Result<(), OutgoingProxyError> {
+        let prepared_socket = if self.non_blocking_tcp_connect
+            && matches!(&request.remote_address, SocketAddress::Ip(..))
+            && request.protocol == NetProtocol::Stream
+        {
+            if let Some(method) = non_blocking_hack::working_method().await {
+                let ipv4 = matches!(&request.remote_address, SocketAddress::Ip(ip) if ip.is_ipv4());
+                Some(method.prepare_socket(ipv4).await?)
+            } else {
+                tracing::warn!(
+                    remote_address = %request.remote_address,
+                    "Cannot emulate non-blocking TCP connect, no working hack detected."
+                );
+                None
+            }
+        } else {
+            None
+        };
+
+        let connection_id = rand::random::<u128>();
```

Comment:
Doesn't random here have a tiny chance of collision? If so, maybe add a comment here?